### PR TITLE
feat(blog): card implemented to blog page

### PR DESCRIPTION
**Pull request by mabsattar**
# Description

 

Added a Card component and implemented the component to all markdown files in the blog directory to display the cards.

## Additional Notes

I had to use URL's for clickmore button, when users click it'll show the full content in a separate page. Also, i modified double quotes with &quot;  in a few to be able to implement the component without any issues.

## Related issue(s)

docs: redesign blog index page #1258

## Confirmation

- [ yes] I have read and understand MACI's [contributor guidelines](https://maci.pse.dev/docs/contributing) and [code of conduct](https://maci.pse.dev/docs/contributing/code-of-conduct).
- [ yes] I have read and understand MACI's [GitHub processes](https://github.com/privacy-scaling-explorations/maci/discussions/847).
- [ yes] I have read and understand MACI's [testing guide](https://maci.pse.dev/docs/testing).

### CODE REVIEW

**Comment by ctrlc03** [comment]
> # Description
> Added a Card component and implemented the component to all markdown files in the blog directory to display the cards.
> 
> ## Additional Notes
> I had to use URL's for clickmore button, when users click it'll show the full content in a separate page. Also, i modified double quotes with " in a few to be able to implement the component without any issues.
> 
> ## Related issue(s)
> docs: redesign blog index page #1258
> 
> ## Confirmation
> * [ yes] I have read and understand MACI's [contributor guidelines](https://maci.pse.dev/docs/contributing) and [code of conduct](https://maci.pse.dev/docs/contributing/code-of-conduct).
> * [ yes] I have read and understand MACI's [GitHub processes](https://github.com/privacy-scaling-explorations/maci/discussions/847).
> * [ yes] I have read and understand MACI's [testing guide](https://maci.pse.dev/docs/testing).

thank you very much for this. Was checking the preview and it looks like the links on the card are not being parsed correctly

Screenshot 2024-06-03 at 09 40 37

File: website/blog/2021-10-12-maci-v1.md [hunk 1]
```diff
@@ -9,6 +9,16 @@ authors:
...
 ---
 
+import React from 'react';
+import Card from '../src/components/BlogCard/Card';
+import styles from '../src/components/BlogCard/blog.module.css';
+
+<div className={styles.cardsGrid}>
+<Card
+  excerpt="The Privacy & Scaling Explorations team is proud to release version 1.0 of Minimal Anti-Collusion Infrastructure (MACI). MACI enables collusion resistance for decentralised applications, particularly voting and quadratic funding systems. This release is a major upgrade to the project and provides better developer experience and gas savings for users."
+  url="https://maci.pse.dev/blog/maci-1-0-release"
```
Comment (ctrlc03) [inline website/blog/2021-10-12-maci-v1.md:19]: could we have relative links instead?

**Review by kittybest — COMMENTED** [review]
I have a question: I think it's good to preview posts while I head to the `/blog` page, but while clicking on a certain post, the full content should be displayed instead of just the preview card (and now if I pressed on the `read more` button, it leads to `maci.pse.dev/blog/[post_name]`, which is not solving this problem (since if we merged this PR, the url `maci.pse.dev/blog/[post_name]` would lead to the blog post page that only shows the card preview, no way to display the full content.

File: website/blog/2021-10-12-maci-v1.md (hunk 1 above)
Comment (mabsattar) [reply website/blog/2021-10-12-maci-v1.md:19]: Hi @ctrlc03  sure, i'm working on it.

**Comment by mabsattar** [comment]
@kittybest  Noted!

**Comment by ctrlc03** [comment]
> @kittybest Noted!

@mabsattar Wanted to check if there was any updates on this? I see that the links are not displayed correctly, and there's also @kittybest suggestion in the comment above! Also, did you have in mind to have the actual index page of the blog present all posts and their preview like some of the examples in the issue (https://mirror.xyz/privacy-scaling-explorations.eth) or did you want us to take care of that part? thanks for your work so far :)

**Comment by mabsattar** [comment]
@ctrlc03 i'm working on it, I haven't pushed the code yet, also i can try and work on the index page as well as mentioned by @kittybest .

**Comment by ctrlc03** [comment]
> @ctrlc03 i'm working on it, I haven't pushed the code yet, also i can try and work on the index page as well as mentioned by @kittybest .

@mabsattar Thank you so much for that! lmk if we can help

**Pull request closed without merging**